core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -374,8 +374,7 @@ private static TableMetadata create(TableOperations ops, UpdateTableRequest requ
     // the only valid requirement is that the table will be created
     request.requirements().forEach(requirement -> requirement.validate(ops.current()));
 
-    TableMetadata.Builder builder = TableMetadata.buildFromEmpty();
-    request.updates().forEach(update -> update.applyTo(builder));
+    TableMetadata.Builder builder = TableMetadata.buildFromMetadataUpdates(request.updates());
```

Comment:
I would just inline the fix here when iterating `updates`

Reply:
Done!